planner/core/logical_plan_builder.go

```diff
@@ -2114,6 +2114,9 @@ func (b *PlanBuilder) buildSelect(ctx context.Context, sel *ast.SelectStmt) (p L
 	if err != nil {
 		return nil, err
 	}
+	if b.capFlag&canExpandAST != 0 {
+		originalFields = sel.Fields.Fields
+	}
```

Comment:
PTAL @zz-jason @winoros

Reply:
Can we add this check for line https://github.com/pingcap/tidb/pull/11818/files#diff-b9e75a99170ea35a67074d5d3d3305adR2241 instead?

Reply:
It seems that we do not need to check it. We only need to set the `originalFields` considering capFlag.